### PR TITLE
[Merged by Bors] - feat: Proof that `IntermediateField.lift` is injective

### PR DESCRIPTION
**Pull request by xroblot**
Add `map_injective` and `lift_injective` for `IntermediateField`. Also minize the imports

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/FieldTheory/IntermediateField.lean [hunk 1]
```diff
@@ -611,6 +609,12 @@ theorem toSubalgebra_injective :
   rw [← mem_toSubalgebra, ← mem_toSubalgebra, h]
 #align intermediate_field.to_subalgebra_injective IntermediateField.toSubalgebra_injective
 
+theorem map_injective {f : L →ₐ[K] L'} (hf : Function.Injective f) :
+    Function.Injective (map f) := by
```
Comment (alreadydone) [inline Mathlib/FieldTheory/IntermediateField.lean:613]: You should be able to use [RingHom.injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Field/Basic.html#RingHom.injective) to remove `hf`

Comment (xroblot) [reply Mathlib/FieldTheory/IntermediateField.lean]: Yes. you're right. Thanks

**Review by alreadydone — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**